### PR TITLE
Add proguard config

### PR DESCRIPTION
**Pull request by IlyaEremin**

### CODE REVIEW

File: app/src/main/java/com/flatstack/android/DI.java [hunk 1]
```diff
@@ -25,38 +22,22 @@ public static File getCacheDir() {
...
-                if (localInstance == null) {
-                    picassoInstance = localInstance = new Picasso.Builder(App.appContext)
-                        .downloader(new OkHttp3Downloader(getHttpClient()))
-                        .build();
-                }
-            }
-        }
-        return localInstance;
+    public static Glide getImageLoader() {
+        return Glide.get(App.appContext);
```
Comment (adelnizamutdinov) [inline app/src/main/java/com/flatstack/android/DI.java]: there's not need for that, Glide.with does not hold the reference to an `Activity`, it's smarter than that :)

**Comment by adelnizamutdinov** [comment]
how can we always be sure that the proguared apk always works?

File: app/build.gradle [hunk 2]
```diff
@@ -47,6 +48,15 @@ android {
 //        }
 //    }
 
+    buildTypes {
+        debug {
+        }
+        release {
+            minifyEnabled true
```
Comment (adelnizamutdinov) [inline app/build.gradle]: are you sure that it's a sensible default?

Comment (IlyaEremin) [inline app/build.gradle]: Yeap, this shrink apk size and early prepare you to 65k limit. It's more complicated to add proguard to existing codebase than using proguard from beginning

**Comment by IlyaEremin** [comment]
I did not investigate tests for proguard. I just launch app and move around screens. Normally, if you does not use reflection and add all suggested proguard configs from libraries app works correctly. 
no reflection -> no problem

**Comment by adelnizamutdinov** [comment]
ok then let's get rid of the glide's singleton and merge

**Comment by adelnizamutdinov** [comment]
:+1:

**Pull request merged**